src/mscorlib/shared/Internal/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -109,6 +109,22 @@ public static int SizeOf<T>()
 #endif
         }
 
+        /// <summary>
+        /// Adds an element offset to the given reference.
+        /// </summary>
+        [Intrinsic]
+        [NonVersionable]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static ref T Add<T>(ref T source, nuint elementOffset)
```

Comment:
This needs to be IntPtr. nuint is bitness specific alias. It should not be used in the public surface.